flake.nix

```diff
@@ -0,0 +1,339 @@
+{
+  description = "ECTester";
+
+  inputs = {
+    nixpkgs.url      = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url  = "github:numtide/flake-utils";
+    gradle2nix.url   = "github:tadfisher/gradle2nix/03c1b713ad139eb6dfc8d463b5bd348368125cf1";
+    # FIXME how to add submodule declaratively?
+    # submodule = {
+    #   url = ./
+    # };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, gradle2nix, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        openssl_3013 = pkgs.openssl.overrideAttrs (_old: rec {
+          version = "3.0.13";
+          pname = "openssl";
+          src = pkgs.fetchurl {
+            url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
+            hash = "sha256-iFJXU/edO+wn0vp8ZqoLkrOqlJja/ZPXz6SzeAza4xM=";
+          };
+          # FIXME this might cause unwanted things
+          patches = [];
+        });
+        openssl_315 = pkgs.openssl.overrideAttrs (_old: rec {
+          version = "3.1.5";
+          pname = "openssl";
+          src = pkgs.fetchurl {
+            url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
+            hash = "sha256-auAVRn2r8EabE5rakzGTJ74kuYJR/67O2gIhhI3AkmI=";
+          };
+          # FIXME this might cause unwanted things
+          patches = [];
+        });
```

Comment:
Are these two blocks necessary? The second one is unused, looks like some leftover. The first one is used though..

Reply:
No, these are leftovers, when I was testing whether changing the version this way works. I was using this for the development environment `nix develop`. Will be removed/refactored.